pages/AlgoDetails/index.js

```diff
@@ -0,0 +1,16 @@
+import { useRouter } from 'next/router';
```

Comment:
I think we should use a `use client` above this import to specify that this file is served on client side. It is a highly recommended implementation in NextJs 13. [This is the reference](https://beta.nextjs.org/docs/routing/linking-and-navigating)